### PR TITLE
UTY-540: ComponentAdded / ComponentRemoved reactive tags

### PR DESCRIPTION
**Pull request by mattyoung-improbable**
#### Description
Created 2 new tag components, (ComponentAdded<T>, ComponentRemoved<T>) that are added to ECS entities for one frame when the SpatialOSComponent T is added / removed.
#### Tests
Tested locally by injecting the ComponentAdded tag into a system and verifying they are present during initial checkout for a single frame.
Used Connection.SendComponentInterest on UnityClient to unregister interest of a SpatialOSComponent and verified the 
#### Documentation
Updated reactive-components.md
#### Primary reviewers
If your change will take a long time to review, you can name at most two primary reviewers who are ultimately responsible for reviewing this request. @ mention them.

### CODE REVIEW

File: workers/unity/Assets/Gdk/Core/Components/ReactiveComponents.cs [hunk 1]
```diff
@@ -28,4 +29,12 @@ public class EventsReceived<T> : MessagesReceived<T> where T : struct, ISpatialE
     public class AuthoritiesChanged<T> : MessagesReceived<Authority>
     {
     }
+
+    public struct ComponentAdded<T> : IComponentData
```
Comment (samiwh) [inline workers/unity/Assets/Gdk/Core/Components/ReactiveComponents.cs]: Can we restrict to to `ISpatialCompoentData`

File: code_generator/Templates/UnityComponentConversionGenerator.tt [hunk 2]
```diff
@@ -127,6 +128,16 @@ namespace <#= qualifiedNamespace #>
                 view.SetComponentObject(entity, <#= componentDetails.CamelCaseTypeName #>);
 <# } #>
                 view.AddComponent(entity, new NotAuthoritative<<#= componentDetails.TypeName #>>());
+
+                if (view.HasComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity))
+                {
+                    view.RemoveComponent<ComponentRemoved<<#= componentDetails.TypeName #>>>(entity);
+                }
+                else if (!view.HasComponent<ComponentAdded<<#= componentDetails.TypeName #>>>(entity))
```
Comment (samiwh) [inline code_generator/Templates/UnityComponentConversionGenerator.tt]: If this is false something has gone wrong. Most likely someone added the component manually.

File: code_generator/End2End/GeneratedDependencies/Generated/improbable/testschema/blittable/BlittableComponentTranslation.cs [hunk 3]
```diff
@@ -123,6 +123,16 @@ public void OnAddComponent(AddComponentOp<global::Improbable.TestSchema.Blittabl
...
                 view.AddComponent(entity, new NotAuthoritative<SpatialOSBlittableComponent>());
+
+                if (view.HasComponent<ComponentRemoved<SpatialOSBlittableComponent>>(entity))
+                {
+                    view.RemoveComponent<ComponentRemoved<SpatialOSBlittableComponent>>(entity);
+                }
+                else if (!view.HasComponent<ComponentAdded<SpatialOSBlittableComponent>>(entity))
+                {
+                    var addedTag = new ComponentAdded<SpatialOSBlittableComponent>();
+                    view.AddComponent(entity, addedTag);
```
Comment (jamiebrynes7) [inline code_generator/End2End/GeneratedDependencies/Generated/improbable/testschema/blittable/BlittableComponentTranslation.cs]: Just inline this?

File: code_generator/Templates/UnityComponentConversionGenerator.tt (hunk 2 above)
Comment (mattyoung-improbable) [reply code_generator/Templates/UnityComponentConversionGenerator.tt]: Added an error log message.

**Pull request merged**